### PR TITLE
Add `NumericFieldStats` for unified numeric field statistics retreival

### PR DESCRIPTION
**Pull request by salvatorecampagna**
## Problem

Lucene has two independent APIs for retrieving the global min/max of a numeric field: `PointValues.getMinPackedValue()` / `PointValues.getMaxPackedValue()` (returns `byte[]` or `null`) and `DocValuesSkipper.globalMinValue()` / `DocValuesSkipper.globalMaxValue()` (returns `long` with sentinel values for "no data"). The sentinel approach is problematic because `Long.MIN_VALUE`/`Long.MAX_VALUE` are legitimate field values, making "no data" indistinguishable from real values.

This caused `SortedNumericDocValuesRangeQuery.rewrite()` to silently skip its optimization when only PointValues were available (no skip index), since the sentinel-based range check could never trigger.

## Solution

`NumericFieldStats` is a utility class that retrieves global field statistics from index metadata structures (`PointValues`, `DocValuesSkipper`) without per-document access. It exposes a single method:

`getStats(IndexReader, String)` returns a `Stats` record `(long min, long max, int docCount)`, or `null` when no metadata structure is available.

It probes `PointValues` first (always available for standard numeric fields, null-based "no data"), then falls back to `DocValuesSkipper` (covers doc-values-only fields with skip index).

`SortedNumericDocValuesRangeQuery.rewrite()` now uses this API, fixing the optimization for fields with PointValues but no skip index.

## Packed value decoding

PointValues stores values as big-endian byte arrays with the sign bit flipped. IntField produces 4-byte arrays, LongField produces 8-byte arrays. The internal `decodeLong` method dispatches on array length to call the right `NumericUtils` decoder. The int case widens to long via Java sign extension, which preserves the value. The API returns long unconditionally because the query layer already works with long bounds internally (even for IntField queries). Callers that need int can safely narrow with `Math.toIntExact()`.

## Why not fix DocValuesSkipper directly?

Changing `DocValuesSkipper.globalMinValue()`/`globalMaxValue()` to return `Long` instead of `long` would fix the sentinel ambiguity at the source. But the per-segment `minValue()`/`maxValue()` methods are called on hot scoring paths where boxing would add overhead, and the static global methods are public API so changing their return type would break existing callers. A higher-level utility avoids both issues.

## Tests

Multiple tests in `TestNumericFieldStats` cover both PointValues and DocValuesSkipper data sources, edge cases (empty index, nonexistent field, doc values without skip index, mixed segments), boundary values, multi-valued fields, and doc count.

An integration test in `TestDocValuesQueries` verifies that `rewrite()` now correctly produces `MatchNoDocsQuery`/`MatchAllDocsQuery` when only PointValues are available.

```
./gradlew -p lucene/core test --tests "org.apache.lucene.search.TestNumericFieldStats"
./gradlew -p lucene/core test --tests "org.apache.lucene.search.TestDocValuesQueries"
```

Closes #15740

### CODE REVIEW

File: lucene/CHANGES.txt [hunk 1]
```diff
@@ -77,6 +77,11 @@ New Features
 Improvements
 ---------------------
 
+* GITHUB#15740: Add NumericFieldStats utility for retrieving global numeric field statistics
```
Comment (romseygeek) [inline lucene/CHANGES.txt]: This can go in the 10.5 section, no need to wait for 11

File: lucene/CHANGES.txt [hunk 2]
```diff
@@ -203,7 +203,11 @@ New Features
 
 Improvements
 ---------------------
-(No changes)
+
```
Comment (romseygeek) [inline lucene/CHANGES.txt:202]: I think I'd put it in `New Features`.  Sorry to be pendantic...

Comment (salvatorecampagna) [reply lucene/CHANGES.txt:202]: Thanks for the suggestion, moved it to New Features. Out of curiosity, what's the rule of thumb for distinguishing New Features from Improvements? This PR adds a new utility class but also fixes an existing optimization, so I wasn't sure where it belongs.

Comment (romseygeek) [reply lucene/CHANGES.txt:202]: I would say 'improvements' are things that happen behind the scenes, so you get to use them just by upgrading.  Whereas 'new features' are new public APIs that you can use (as distinguished from 'API changes' which are things that you need to alter in your code so that it still compiles).

**Review by romseygeek — APPROVED** [review]

File: lucene/CHANGES.txt (hunk 2 above)
Comment (romseygeek) [reply lucene/CHANGES.txt:202]: Oh, but then I'd missed the fact that this actually fixes a missing opto in the range query.  Eh, so it could go in either, but I think it's still better highlighted in the 'New Features' section.

Comment (salvatore-campagna) [reply lucene/CHANGES.txt:202]: Let's put it this way: the original intent was to extract a common API for min/max (new feature). The optimization of the range query when fields have point values is a nice side effect as a consumer of the new API.

**Comment by romseygeek** [comment]
Thanks @salvatore-campagna!
